pkg/registry/discovery/endpointslice/strategy.go

```diff
@@ -186,11 +188,40 @@ func dropTopologyOnV1(ctx context.Context, oldEPS, newEPS *discovery.EndpointSli
 			return
 		}
 
+		// Only node names that exist in previous version of the EndpointSlice
+		// deprecatedTopology fields may be retained in new version of the
+		// EndpointSlice.
+		prevNodeNames := getDeprecatedTopologyNodeNames(oldEPS)
+
 		for i := range newEPS.Endpoints {
 			ep := &newEPS.Endpoints[i]
 
-			//Silently clear out DeprecatedTopology
-			ep.DeprecatedTopology = nil
+			newTopologyNodeName, ok := ep.DeprecatedTopology[corev1.LabelHostname]
+			if ep.NodeName == nil && ok && prevNodeNames.Has(newTopologyNodeName) {
+				// Retain node name in deprecatedTopology until the node name field has been populated
+				ep.DeprecatedTopology = map[string]string{corev1.LabelHostname: newTopologyNodeName}
+			} else {
+				// Silently clear out DeprecatedTopology
```

Comment:
This comment sounds so suspicious to me (Silently clear out ...)

Reply:
how about:
```
// drop writes to this field via the v1 API as documented
```

Reply:
Updated, thanks!